PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx

```diff
@@ -164,6 +195,92 @@ struct HfCandidateSelectorLcToK0sP {
   }
 
+  template <typename T, typename U>
+  bool selectionML(const T& hfCandCascade, const U& bach)
+  {
+
+    auto candPt = hfCandCascade.pt();
+    int ptBin = findBin(binsPtML, candPt);
+    if (ptBin == -1) {
+      return false;
+    }
+
+    float inputParamsAll[24] = {
+      hfCandCascade.rSecondaryVertex(),
+      hfCandCascade.decayLength(),
+      hfCandCascade.ptProng0(),
+      hfCandCascade.ptProng1(),
+      hfCandCascade.impactParameter0(),
+      hfCandCascade.impactParameter1(),
+      hfCandCascade.v0radius(),
+      hfCandCascade.v0cosPA(),
+      hfCandCascade.mLambda(),
+      hfCandCascade.mAntiLambda(),
+      hfCandCascade.mK0Short(),
+      hfCandCascade.mGamma(),
+      o2::aod::hf_cand_casc::ctV0K0s(hfCandCascade),
+      o2::aod::hf_cand_casc::ctV0Lambda(hfCandCascade),
+      hfCandCascade.dcaV0daughters(),
+      static_cast<float>(hfCandCascade.ptV0Pos()),
+      hfCandCascade.dcapostopv(),
+      static_cast<float>(hfCandCascade.ptV0Neg()),
+      hfCandCascade.dcanegtopv(),
+      bach.tpcNSigmaPr(),
+      bach.tofNSigmaPr(),
+      hfCandCascade.cpa(),
+      o2::aod::hf_cand_3prong::ctLc(hfCandCascade),
+      o2::aod::hf_cand_3prong::yLc(hfCandCascade)};
+
+    std::vector<float> inputFeaturesF;
+    std::vector<double> inputFeaturesD;
+    uint32_t i = 1;
+    for (const auto& param : inputParamsAll) {
+      if (i & MLInputs) {
+        inputFeaturesF.push_back(param);
+        inputFeaturesD.push_back(param);
+      }
+      i = i * 2;
+    }
+
+    float scores[3] = {-1.f, -1.f, -1.f};
+    if (dataTypeML == 1) {
+      auto scoresRaw = model.evalModel(inputFeaturesF);
+      for (int iScore = 0; iScore < 3; ++iScore) {
+        scores[iScore] = scoresRaw[iScore];
+      }
+    } else if (dataTypeML == 11) {
+      auto scoresRaw = model.evalModel(inputFeaturesD);
+      for (int iScore = 0; iScore < 3; ++iScore) {
+        scores[iScore] = scoresRaw[iScore];
+      }
+    } else {
+      LOG(error) << "Error running model inference for Lc: Unexpected input data type.";
+    }
+
+    registry.fill(HIST("hBDTScoreBkg"), scores[0]);
+    registry.fill(HIST("hBDTScoreBkgVsPtCand"), scores[0], candPt);
+    registry.fill(HIST("hBDTScorePrompt"), scores[1]);
+    registry.fill(HIST("hBDTScorePromptVsPtCand"), scores[1], candPt);
+    registry.fill(HIST("hBDTScoreNonPrompt"), scores[2]);
+    registry.fill(HIST("hBDTScoreNonPromptVsPtCand"), scores[2], candPt);
+
+    if (scores[0] > thresholdBDTScore->get(ptBin, "BDTbkg")) {
+      return false;
+    }
+    if (scores[1] <= thresholdBDTScore->get(ptBin, "BDTprompt") && scores[2] <= thresholdBDTScore->get(ptBin, "BDTnonprompt")) {
+      return false;
+    }
+    /*
+    if (scores[1] > thresholdBDTScore->get(ptBin, "BDTprompt")) {
+      // prompt
+    }
+    if (scores[2] > thresholdBDTScore->get(ptBin, "BDTnonprompt")) {
+      // non-prompt
+    }
+    */
```

Comment:
Is this needed?

Reply:
No, it was just meant to be as a placeholder for later, because for the moment there is no distinction between 'prompt' and 'nonprompt'. It only returns 'false' for background and 'true' for signal (prompt or nonprompt). I took it out